content/types/_channel_event.textile

```diff
@@ -1,25 +1,92 @@
-@ARTChannelEvent@ is the enum emitted as the event in @ARTRealtimeChannel.on@. The variants correspond to "@Realtime Channel@ state":/realtime/channels-messages/#channel-state changes, plus an arbitrary @Error@.
-
-```[objc]
-  typedef NS_ENUM(NSUInteger, ARTChannelEvent) {
-      ARTChannelEventInitialized,
-      ARTChannelEventAttaching,
-      ARTChannelEventAttached,
-      ARTChannelEventDetaching,
-      ARTChannelEventDetached,
-      ARTChannelEventFailed,
-      ARTChannelEventError
-  };
-```
-
-```[swift]
-  enum ARTChannelEvent : UInt {
-      case Initialized
-      case Attaching
-      case Attached
-      case Detaching
-      case Detached
-      case Failed
-      case Error
-  }
-```
+blang[jsall].
+  @ChannelEvent@ is a String that can be emitted as an event on the @Channel@ object; either a "@ChannelState@":/realtime/channels-messages#channel-states or an @update@ event.
+
+
+  ```[javascript]
+    var allChannelEvents = allChannelStates.concat(['update']);
```

Comment:
It's not defined in this partial, but everywhere `channel_event` is included, `channel_state` is included right above it, which defines it. I figured this way makes it clearer what the difference actually is between the two. But happy to switch out for a full list if you think that'd be better, wdyt?